pkg/component/pushed_component.go

```diff
@@ -143,22 +142,24 @@ func (d devfileComponent) GetEnvVars() []v12.EnvVar {
 func (d devfileComponent) GetLabels() map[string]string {
 	return d.d.Labels
 }
-
 func (d devfileComponent) GetAnnotations() map[string]string {
 	return d.d.Annotations
 }
 
 func (d devfileComponent) GetName() string {
-	return d.d.Labels[componentlabels.ComponentLabel]
+	return d.d.Labels[componentlabels.KubernetesInstanceLabel]
 }
 
 func getType(component provider) (string, error) {
-	if componentType, ok := component.GetAnnotations()[componentlabels.ComponentTypeAnnotation]; ok {
+
+	// For backwards compatibility with previously deployed components that could be non-odo, check the label first
+	// then check to see if there is an annotation with the project type
+	if componentType, ok := component.GetLabels()[componentlabels.OdoProjectTypeAnnotation]; ok {
```

Comment:
Wouldn't it be better to do it another way around? First check annotations and then labels. Annotations is what we want to use primarily for this.